mne/inverse_sparse/mxne_optim.py

```diff
@@ -372,13 +414,46 @@ def _mixed_norm_solver_bcd(M, G, alpha, lipschitz_constant, maxit=200,
                              % (gap, tol))
                 break
 
+        # using Anderson acceleration of the primal variable for faster
+        # convergence
+        if use_accel:
+            last_K_X[i % (K + 1)] = X
+
+            if i % (K + 1) == K:
+                for k in range(K):
+                    U[k] = last_K_X[k + 1].ravel() - last_K_X[k].ravel()
+                C = U @ U.T
+                one_vec = np.ones(K)
+
+                try:
```

Comment:
the `try` part should include the minimal set of lines to catch a given exception. In this case it looks like only the `np.linalg.solve` line should be inside the try/except. The other lines can go in an `else` clause (which occurs if no exception is hit)

Moreover, when there is an error, it silently just continues as if there wasn't a problem. This is abnormal -- usually we emit warnings in these cases -- so if it's really okay to silently ignore this in this case, some code comment saying why would be good, e.g.:
```
# This code is just an attempt at a speedup, so if it fails we can
# safely continue using the non-sped-up code
```

Reply:
I'll change that. Indeed, it is ok to have a warning in `np.linalg.solve` sometimes. Sometimes, the matrix U is singular by construction and can't be inverted, hence the `try` part.